LFS-731: Support more than 10 forms

acrowthe opened the pull request:
Add support for arbitrary numbers of forms to the user dashboard
Add paginated table to NewFormDialog

Test branch with 16 forms available [here](https://github.com/ccmbioinfo/lfs/tree/LFS-731-test)

acrowthe committed:
LFS-731: Support more than 10 forms
Add support for arbitrary numbers of forms to the user dashboard
Add paginated table to NewFormDialog

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/NewFormDialog.jsx:
```diff
@@ -232,27 +221,48 @@ function NewFormDialog(props) {
+                  search: true,
+                  actionsColumnIndex: -1,
+                  addRowPosition: 'first',
+                  pageSize: 10,
```
For consistency with the subjects table, I suggest to use a page size of 5.
```suggestion
                  pageSize: 5,
```

acrowthe replied:
MaterialTable defaults to 5 results: removed pageSize option to fallback to that default as that is what the subject table is doing.

marta- reviewed (changes requested):
Search doesn't appear to be working:
![captured (11)](https://user-images.githubusercontent.com/651980/100016414-60196f00-2da7-11eb-928a-be83ca98ecf1.gif)

acrowthe committed:
LFS-731: Support more than 10 forms
- Fix newFormDialog form page search and default result count

acrowthe commented:
Fixed search

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/UserDashboard.jsx:
```diff
@@ -66,13 +67,18 @@ function UserDashboard(props) {
 
     // Fetch the questionnaires
-    fetch("/query?query=select * from [lfs:Questionnaire]")
+    fetch(`/query?limit=${numberForms || 11}&query=select * from [lfs:Questionnaire]`)
```
Why the magic number 11?

acrowthe replied:
Remnant of a test. Refactored to remove magic number

acrowthe committed:
LFS-731: Support more than 10 forms
Remove magic number

IntegralProgrammer commented on modules/data-entry/src/main/frontend/src/dataHomepage/NewFormDialog.jsx:
```diff
@@ -232,27 +221,47 @@ function NewFormDialog(props) {
+                    )
+                  })
+                }}
+                onRowClick={(event, rowData) => {
+                  if (!isFetching) {
+                    setSelectedQuestionnaire(rowData);}}
+                  }
```
```suggestion
                onRowClick={(event, rowData) => {
                  if (!isFetching) {
                    setSelectedQuestionnaire(rowData);
                  }
                }}
```

IntegralProgrammer commented:
I have found a bug that occurs when creating a new form.

1. Go to the *Forms* page
2. Click the blue **+** icon to create a new Form
3. On the modal that pops-up, switch from _5 rows_ to _10 rows_
4. Click on a Form type (eg. _Somatic Variants Data_)
5. The modal (unexpectedly) switches back from displaying _10 rows_ to displaying _5 rows_

acrowthe commented:
> I have found a bug that occurs when creating a new form.
> 
> 1. Go to the _Forms_ page
> 2. Click the blue **+** icon to create a new Form
> 3. On the modal that pops-up, switch from _5 rows_ to _10 rows_
> 4. Click on a Form type (eg. _Somatic Variants Data_)
> 5. The modal (unexpectedly) switches back from displaying _10 rows_ to displaying _5 rows_

I tested this with the subject selector, and it seems to have the same issue as well, since they both use MaterialTable in a similar way. @marta- I'll fix the form issue here, and then make a new issue+PR to fix the subject selector, if that sounds good to you.

sashaandjic reviewed (approved):
tested - the only remaining issue is the one mentioned above

sashaandjic commented:
Search still does not work for 1 character. For example when I type "T", nothing happens:
![image](https://user-images.githubusercontent.com/23243244/100410417-9facb900-303c-11eb-9524-130b490303b3.png)
Typing "Tu" will give the correct result:
![image](https://user-images.githubusercontent.com/23243244/100410437-ba7f2d80-303c-11eb-8fad-9391087f8bb0.png)

sashaandjic commented:
Problem found.
Steps to reproduce:

- create patient
- create Participant Status form for the patient
- now try to create Demographic data form for the same patient
- an error msg appears **correctly** "Patient P1 already has 1 Demographic data form(s) filled out."
- click on the BACK bttn
- now try to select another form
- nothing happens - **wrong**
- click on CANCEL bttn
- click on + bttn to create a new form
- dialog opens up but it is unresponsive - **wrong**

acrowthe committed:
LFS-731: Support more than 10 forms
- Address code style comment
- Fix buggy table page size
- Fix inablity to progress if backed out of invalid selected subject
- Limit form search to title to prevcent unexpected matches

acrowthe commented:
> > I have found a bug that occurs when creating a new form.
> > 
> > 1. Go to the _Forms_ page
> > 2. Click the blue **+** icon to create a new Form
> > 3. On the modal that pops-up, switch from _5 rows_ to _10 rows_
> > 4. Click on a Form type (eg. _Somatic Variants Data_)
> > 5. The modal (unexpectedly) switches back from displaying _10 rows_ to displaying _5 rows_
> 
> I tested this with the subject selector, and it seems to have the same issue as well, since they both use MaterialTable in a similar way

#371 created to fix this issue on the subject selection page

sashaandjic commented:
Problem that Mike reported is not yet fixed.

sashaandjic reviewed (approved):
tested; both issues are fixed now
